apps/pwabuilder/src/script/pages/app-report.ts

```diff
@@ -1617,18 +1716,93 @@ export class AppReport extends LitElement {
     return chosenColor
   }
 
+  private async applyManifestContext(url: string, manifestUrl?: string, manifestRaw?: string) {
+    this.manifestContext = await processManifest(url, {url: manifestUrl, raw: manifestRaw});
+    this.createdManifest = this.manifestContext.isGenerated || false;
+    setManifestContext(this.manifestContext);
+    this.isAppCardInfoLoading = false;
+    await this.populateAppCard(this.manifestContext, manifestUrl);
+  }
+
   // Runs the Manifest, SW and SEC Tests. Sets "canPackage" to true or false depending on the results of each test
   async runAllTests(url: string) {
     this.runningTests = true;
-    await this.getManifest(url);
-    await Promise.all([ this.testManifest(), this.testSecurity(url)]).then(() =>
+    this.isAppCardInfoLoading = true;
+
+    let findersResults = {
+      manifest: {} as {url?: string, raw?: string, json?: unknown},
+      serviceWorker: {} as {url?: string, raw?: string}
+    }
+
+    // Take only good results, ignore errors.
+    FindWebManifest(url).then( async (result) => {
+      if (result?.content?.raw) {
+        // TODO: can use json instead of raw
+        findersResults.manifest = result.content;
+        await this.applyManifestContext(url, result?.content?.url || undefined, result?.content?.raw);
+        await this.testManifest();
+      }
+    });
+    // .catch(async () => {
+    //   await this.applyManifestContext(url, undefined, undefined);
+    // }).finally(async () => {
+
+    //   await this.testManifest();
+    // });
+
+    FindServiceWorker(url).then( async (result) => {
+        if (result?.content?.url) {
+          await AuditServiceWorker(result.content.url).then( (result) => {
+            this.testServiceWorker(processServiceWorker(result.content, undefined));
+          });
+          findersResults.serviceWorker = result.content;
+        }
+        // else {
+        //   this.testServiceWorker(processServiceWorker({score: false, details: {}}, false));
+        // }
+      }
+    );
+    // .catch(() => {
+    //   this.testServiceWorker(processServiceWorker({score: false, details: {}}, false));
+    // });
+
+    try {
+      this.reportAudit = await Report(url);
+    } catch (e) {
+      console.error(e);
+      await this.testSecurity(processSecurity());
+      if (!findersResults.manifest?.raw) {
+        await this.applyManifestContext(url, undefined, undefined);
+        await this.testManifest();
+      }
+      if (!findersResults.serviceWorker?.raw) {
+        this.testServiceWorker(processServiceWorker({score: false, details: {}}, false));
+      }
+      this.runningTests = false;
+      return;
+    }
+
+    console.log(this.reportAudit);
+
+    if (this.reportAudit?.artifacts?.webAppManifest) {
```

Comment:
Before overwriting the first manifest with the new one, we should compare them to make sure we aren't wasting cycles overwriting the same infomration over itself.